lib/public_key/src/pubkey_policy_tree.erl

```diff
@@ -0,0 +1,409 @@
+    lists:filtermap(fun(#{valid_policy := ?anyPolicy}) ->
+                            false;
+                       (Node) ->
+                            {true, Node}
+                    end, Children);
+constrain(_) ->
+    [].
+
+
+filter_leafs({Parent0,[{_, _}|_] = Children0}, LeafFun) ->
+    {Parent0, lists:map(fun(Children)->
+                                filter_leafs(Children, LeafFun)
+                              end, Children0)};
+filter_leafs({Parent, Children}, LeafFun) ->
+    {Parent, lists:filtermap(LeafFun, Children)}.
+
+find_any_leaf(Children, N) when  N > 0->
+    find_any_leaf(Children, N-1);
+find_any_leaf([{_, Children}], 0) ->
+    find_any_leaf(Children);
+find_any_leaf(Children, 0) ->
+    find_any_leaf(Children).
+
+find_any_leaf([]) ->
+    no_node;
+find_any_leaf([#{valid_policy := ?anyPolicy} = Node| _]) ->
+    Node;
+find_any_leaf([_ | Rest]) ->
+    find_any_leaf(Rest).
+
+format_qualifers(asn1_NOVALUE) ->
+    [];
+format_qualifers(Value) ->
+    Value.
+
+keep_policy_node(#{valid_policy := ?anyPolicy} = Node, _) ->
+    {true, Node};
+keep_policy_node(Node, InvalidNodes) ->
+    case lists:member(Node, InvalidNodes) of
+        true ->
+            false;
+        false ->
+            {true, Node}
+    end.
+
+prune_children(TreeBottom) when is_list(TreeBottom) ->
+    lists:filtermap(fun({Parent, Children}) ->
+                             case prune_nodes(Parent, Children) of
+                                 {Parent, []} ->
+                                     false;
+                                 {Parent, NewChildren} ->
+                                     {true, {Parent, NewChildren}}
+                             end;
+                       (Children) ->
+                            {true, Children}
+                    end, TreeBottom).
+
+prune_nodes(Parent, Children) ->
+    {Parent, prune_children(Children)}.
+
+prune_invalid_nodes_children(TreeBottom, InvalidNodes) when is_list(TreeBottom)->
+    lists:filtermap(fun({Parent, Children}) ->
+                            case keep_policy_node(Parent, InvalidNodes) of
+                                false -> %% Prune branch
+                                    false;
+                                {true, #{valid_policy := ?anyPolicy} = Parent} -> %% Keep Parent
+                                    {true, {Parent, prune_invalid_nodes_children(Children, InvalidNodes)}};
+                                {true, Parent} ->
+                                    {true, {Parent, Children}} %% Keep branch
+                            end;
+                       (#{} = Child) -> % Possibly prune leaf
+                            keep_policy_node(Child, InvalidNodes)
+                    end, TreeBottom).
+
+workaround_to_long_notice(Qualifier) ->
+    %% RFC 3280 states that certificate users SHOULD gracefully handle
+    %% explicitText with more than 200 characters.
+    try public_key:der_decode('OTPUserNotice', Qualifier) of
```

Comment:
It is the workaround. We will allow max 350 chars (test case has 310). And if it is longer than that we
will provide a sorry it was to long 'UserNotice'.  200 is the real max.

Reply:
- see https://dictionary.cambridge.org/dictionary/english/workaround
- IMHO, workaround suggests that there is a bug or misbehavior in code - and solution does not completely solve it
- if this solution seems complete and permanent (asn1 behaves properly, but RFC relaxes the rules a little)
  - we are not waiting for some asn1 fix
- then I think it would be better to name this part `handle_longer_notice` not using wording `workaround`
- and also adding a small comment explaining asn1 related behavior here (not only RFC req)

Reply:
It is a workaround for misbehaving encoders, so not a misbehavior in our code. But I can call it handle if it makes you happy :)  I also updated the comment a little.